python/ray/dag/dag_node.py

```diff
@@ -84,14 +91,27 @@ def _collect_upstream_nodes(self) -> List["DAGNode"]:
         them up instead of reference counting. We should consider using weak references
         to avoid circular references.
         """
+        upstream_nodes: List["DAGNode"] = []
+
+        assert hasattr(self._bound_args, "__iter__")
+        for arg in self._bound_args:
+            if isinstance(arg, DAGNode):
+                upstream_nodes.append(arg)
+            else:
+                scanner = _PyObjScanner()
+                dag_nodes = scanner.find_nodes([arg])
```

Comment:
Do you need to wrap `arg` in a list?

Reply:
updated https://github.com/ray-project/ray/pull/48302/commits/adace80b33d0d789185204ab103f899cca0affea